association_backend/association/webhooks/views.py

```diff
@@ -0,0 +1,124 @@
+        return AssociationRepository(
+            session=cast(AsyncSession, get_session(get_engine(echo=False)))
+        )  # session=AsyncSession
+
+    async def handle_customer_subscription_created(self, request, payload):
+        print(f"subscription_created : {payload}")
+        stripe_obj = payload["object"]
+        try:
+            await services.set_subscription_payed(
+                SubscriptionInputModel(subscription_id=stripe_obj["id"]),
+                association_repository=self._get_association_repository(request),
+            )
+            return JSONResponse({"event": payload.get("type", ""), "msg": "OK"})
+        except SubscriptionNotCreated as ex:
+            return JSONResponse(
+                {"event": payload.get("type", ""), "msg": "KO", "error": str(ex)}
+            )
+
+    async def handle_checkout_session_completed(self, request, payload):
+        stripe_obj = payload["object"]
+        try:
+            await services.update_draft_subscription(
+                SubscriptionUpdateInput(
+                    session_id=stripe_obj["id"],
+                    customer_id=stripe_obj["customer"],
+                    subscription_id=stripe_obj["subscription"],
+                ),
+                association_repository=self._get_association_repository(request),
+            )
+            return JSONResponse({"event": payload.get("type", ""), "msg": "OK"})
+        except SubscriptionNotUpdated as ex:
+            return JSONResponse(
+                {"event": payload.get("type", ""), "msg": "KO", "error": str(ex)}
+            )
+
+    async def handle_customer_subscription_updated(self, request, payload):
+        print(f"subscription_updated : {payload}")
+        return JSONResponse({"event": payload.get("type", ""), "msg": "OK"})
+
+    async def handle_payout_paid(self, request, payload):
+        return JSONResponse({"event": payload.get("type", ""), "msg": "KO"})
+
+    async def post(self, request):
+        webhook_secret = STRIPE_WEBHOOK_SIGNATURE_SECRET
+        request_data = await request.body()
+        logger.debug(f"request_data : {request_data}")
+
+        if webhook_secret:
+            # Retrieve the event by verifying the signature using the raw body and secret if webhook signing is configured.
+            signature = request.headers.get("stripe-signature")
+            logger.debug(f"signature : {signature}")
+            try:
+                event = stripe.Webhook.construct_event(
+                    payload=request_data, sig_header=signature, secret=webhook_secret
+                )
+                data = event["data"]
+            except Exception as e:
+                logger.exception(str(e))
+                raise e
+            # Get the type of webhook event sent - used to check the status of PaymentIntents.
+            event_type = event["type"]
+        else:
+            data = request_data["data"]
+            event_type = request_data["type"]
```

Comment:
if there is not webhook_secret we shouldn't go ahead?

Reply:
I copied it from their shared example, I think so
I go deeper into the matter

Reply:
Now I have raised a `WebhookSecretMissing` Exception
I took it from https://github.com/stripe-samples/checkout-single-subscription/blob/master/server/python/server.py